### DISCUSSION TITLE
Ethernet driver for DP83826IRHBR in STM32 nucleo_H743zi in MII mode

### DISCUSSION THREAD

**Discussion by tejueaton** [zephyrproject-rtos/zephyr#65308] [Q&A]
We are designing the custom board using stm32 nucleoH743zi, for that we have used DP83826IRHBR PHY chip in MII mode.
But we are not able to up the ethernet. We are using static IP address.

Below are the configuartion i have made,

#Ethernet MAC support
CONFIG_ETH_DRIVER=y
CONFIG_NET_L2_ETHERNET=y
#Mostly, this is ethernet MAC I need use
#But there is no way to plugin phy
#so we may need patch the STM32 driver
CONFIG_ETH_STM32_HAL=y
#Enable MII interface for STM32 PHY. This will automatically enable MII
CONFIG_ETH_STM32_HAL_MII=y
#Conduct auto PHY speed auto negotiation
#Need Alert patch
CONFIG_ETH_STM32_AUTO_NEGOTIATION_ENABLE=y
#100M or 10M, default is 100M
CONFIG_ETH_STM32_SPEED_10M=y
#CONFIG_NET_PROMISCUOUS_MODE=y
CONFIG_ETH_STM32_HAL_PHY_ADDRESS=1

And in dts file we have edit this for MII connection.
&mac {
	status = "okay";
	pinctrl-0 = <&eth_mdc_pc1
		     &eth_rxd0_pc4
		     &eth_rxd1_pc5  
			 &eth_rxd2_pb0    
			 &eth_rxd3_pb1
		     &eth_ref_clk_pa1
		     &eth_mdio_pa2
		     &eth_rx_dv_pa0
		     &eth_tx_en_pb11
		     &eth_txd0_pb12
		     &eth_txd1_pb13
			 &eth_txd2_pc2
			 &eth_txd3_pe2>;
	pinctrl-names = "default";
};

So zephyr has supporting driver for this particular PHY chip.
If yes, what are the necessary configuration needs to done.

**Comment by nordicjm** [comment]
Can you edit this to include the required information from the bug template?

**Comment by henrikbrixandersen** [comment]
Please use our bug template when reporting bugs. You need to edit this issue to include the information requested in https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/ISSUE_TEMPLATE/001_bug_report.md

**Comment by erwango** [comment]
> So zephyr has supporting driver for this particular PHY chip.

It doens't look it is supported

**Comment by mariopaja** [comment]
Which Zephyr version are you using? 

In my case, we are using a custom rt1064 board with an TI 100base-t1 SPE chip and until zephyr v3.3.0 we could use it directly with the with mcux drivers without the need of extra support for that chip. However, I haven't been able to use it with the latest zephyr version and I would have to investigate what exactly is changed.

I dont know exactly stm ethernet driver works but maybe testing other zephyr versions can help
